packages/query-store/src/QueryStore.ts

```diff
@@ -347,13 +353,24 @@ export class QueryStore extends AbstractStore<QueryStoreValue> {
 				this.#mockResult = Object.fromEntries(this.#columns.map((c) => [c.name, null]));
 				this.#metaLoading = false;
 				this.#metaLoaded = true;
+				this.#calculateScore();
 				this.publish();
 			},
 			() => this.#exec(`--col-metadata\nDESCRIBE ${this.#query.toString()}`),
 			this.#setError
 		);
 	};
 
+	#calculateScore = () => {
```

Comment:
This function feels a little odd to me; I don't expect something called `calculateScore ` to notify the user; might want to split this into 2 functions

1 should calculate and return the score; the other should be responsible for the notification 

We can also expose the score once it's split up